### PR TITLE
Properly creating arguments to pass to setup.py

### PR DESCRIPTION
**Pull request by davidbrownellWork**
**Description**: 
Fixed issue where args were not being properly passed to setup.py.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by jywu-msft** [comment]
2020-04-29 06:17:04,768 Build [DEBUG] - Subprocess completed. Return code=0
2020-04-29 06:17:04,769 Build [DEBUG] - Running subprocess in '/build/Release'
['/usr/bin/python3', '/onnxruntime_src/setup.py', 'bdist_wheel', '--use_dnnl', '--wheel_name_suffix=featurizers', '--use_featurizers']
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help

error: option --use_featurizers not recognized

**Comment by jywu-msft** [comment]
if --use_dnnl and --use_featurizers options can be enabled simultaneously in build.py 
then --use_featurizers check in https://github.com/microsoft/onnxruntime/blob/master/setup.py#L50
cannot be an elif

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1360,9 +1360,9 @@ def build_python_wheel(
             args.append('--use_nuphar')
         if wheel_name_suffix:
             args.append('--wheel_name_suffix={}'.format(wheel_name_suffix))
-        elif use_acl:
+        if use_acl:
```
Comment (jywu-msft) [inline tools/ci_build/build.py:1363]: I think use_acl should be moved below use_nuphar as elif

File: tools/ci_build/build.py [hunk 2]
```diff
@@ -1360,9 +1360,9 @@ def build_python_wheel(
...
             args.append('--use_acl')
-        elif featurizers_build:
+        if featurizers_build:
```
Comment (jywu-msft) [inline tools/ci_build/build.py:1365]: to mirror this change,
I think https://github.com/microsoft/onnxruntime/blob/master/setup.py#L50 
should be changed to if (from elif)

File: tools/ci_build/build.py (hunk 1 above)
Comment (davidbrownellWork) [reply tools/ci_build/build.py]: Reconciled the argument usage between `setup.py` and `build.py`, however `build.py` is passing `--use_acl` to `setup.py`, which isn't explicitly parsed in that file.

File: tools/ci_build/build.py (hunk 2 above)
Comment (davidbrownellWork) [reply tools/ci_build/build.py]: Agreed and updated.

**Comment by davidbrownellWork** [comment]
Thanks for taking a look at these changes, @jywu-msft. I wasn't quite sure how to reason about what was mutually exclusive and what could be freely provided.

**Pull request merged**